src/utils/retry.test.ts

```diff
@@ -0,0 +1,33 @@
+import { retry } from './retry'
+
+describe('retry function', () => {
+  jest.useRealTimers()
+  let mockFn: jest.Mock
+
+  beforeEach(() => {
+    mockFn = jest.fn()
+  })
+
+  it('should resolve when function is successful', async () => {
+    const expectedResult = 'Success'
+    mockFn.mockResolvedValue(expectedResult)
+    const result = await retry(mockFn)
+    expect(result).toEqual(expectedResult)
+    expect(mockFn).toHaveBeenCalledTimes(1)
+  })
+
+  it('should retry the specified number of times before rejecting', async () => {
+    const error = new Error('Failure')
+    mockFn.mockRejectedValue(error)
+    await expect(retry(mockFn, 3, 1)).rejects.toEqual(error)
```

Comment:
Check that this takes at least 1 + 2 + 4 = 7 ms.

Reply:
there isn't really a way to do this without using fake timers, which we haven't figured out yet

Reply:
you can do this by just checking `Date.now()` before and after the call to `retry`, and ensuring the difference is >= 7.